internal/runners/history/history.go

```diff
@@ -57,7 +60,25 @@ func (h *History) Run(params *HistoryParams) error {
 		if h.project == nil {
 			return locale.NewInputError("err_history_no_project", "A namespace was not provided and a project could not be found. Please use a project namespace or run this command in a project directory")
 		}
-		commits, err = model.CommitHistoryFromID(h.project.CommitUUID())
+
+		defBranch, err := model.DefaultBranchForProjectName(h.project.Owner(), h.project.Name())
+		if err != nil {
+			return fmt.Errorf("branch: %w", err)
+		}
+
+		if defBranch.Label != h.project.BranchName() {
+			return fmt.Errorf("remote branch not same as local")
+		}
```

Comment:
I'm not sure what this is meant to achieve. We of course want to support the use-case of someone not being on the default branch..